src/Compression/ICompressionCodec.h

```diff
@@ -96,6 +98,11 @@ class ICompressionCodec : private boost::noncopyable
     /// Read size of decompressed block from compressed source
     UInt32 readDecompressedBlockSize(const char * source) const;
 
+    virtual void setDimensions(const std::vector<size_t> & /*dimensions*/)
```

Comment:
It is not clear for the average reader what a "dimension" is in the context of compression codecs. Needs a code comment. If you think "dimensions" will ever only be useful for SZ3, please rename to something like "setDimensionsForSZ3".